sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -298,3 +299,56 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 
   override def prettyName: String = "length"
 }
+
+/**
+ *  A function that return soundex code of the given string expression.
+ */
+case class SoundEx(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  @transient lazy val so = new Soundex
+
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val string = child.eval(input)
+    if (string == null) {
+      null
+    } else {
+      val str: String = string.toString
+      UTF8String.fromString(so.encode(str))
+    }
+  }
+}
+
+/**
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
+ */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  @transient lazy val so = new Soundex
+
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def eval(input: InternalRow): Any = {
```

Comment:
i just tried to implement one in UTF8String. The difficulty is i need to enum all specific alphabets used in Europe. Though converting the uppercase by char ascii numbers works, i'm not sure if i list all the available alphabets correctly. i've tested in scala console, the toUpper of stringbuffer can convert the specific char(s). 

test case
scala> val sb = new StringBuffer()
sb: StringBuffer = 

scala> val string = 'δ'
string: Char = δ

scala> sb.append(string)
res0: StringBuffer = δ

scala> sb.charAt(0).toUpper
res1: Char = Δ